src/Swarm/Game/Scenario.hs

```diff
@@ -134,7 +134,7 @@ instance FromJSONE EntityMap Scenario where
         <*> pure known
         <*> localE (,rsMap) (v ..: "world")
         <*> pure rs
-        <*> liftE (v .:? "objectives" .!= [])
+        <*> pure objectives
```

Comment:
Why not:
```suggestion
        <*> liftE (v .:? "objectives" .!= []) >>= validateObjectives
```
(disclaimer: not tested)

Reply:
There must be some operator that would work here, but as is this doesn't typecheck.